CSharpSourceCode/Battle/TriggeredEffect/Scripts/SummonSkeleton.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using TaleWorlds.Core;
+using TaleWorlds.Library;
+using TaleWorlds.Localization;
+using TaleWorlds.MountAndBlade;
+using TaleWorlds.ObjectSystem;
+
+namespace TOW_Core.Battle.TriggeredEffect.Scripts
+{
+    public class SummonSkeleton : ITriggeredScript
+    {
+        public void OnTrigger(Vec3 position, Agent triggeredByAgent)
+        {
+            SpawnAgent(triggeredByAgent, position);
+        }
+
+        private void SpawnAgent(Agent caster, Vec3 position)
+        {
+            BasicCharacterObject troopCharacter = MBObjectManager.Instance.GetObject<BasicCharacterObject>("tow_skeleton_recruit");
+            CustomBattleTroopSupplier supplier = new CustomBattleTroopSupplier((CustomBattleCombatant)caster.Origin.BattleCombatant, !caster.Team.IsEnemyOf(Mission.Current.PlayerTeam));
+            CustomBattleAgentOrigin troopOrigin = new CustomBattleAgentOrigin((CustomBattleCombatant)caster.Origin.BattleCombatant, troopCharacter, supplier, !caster.Team.IsEnemyOf(Mission.Current.PlayerTeam));
+            MatrixFrame frame = new MatrixFrame(Mat3.Identity, position);
+            AgentBuildData buildData = new AgentBuildData(troopCharacter).
+                Team(caster.Team).
+                ClothingColor1(caster.Team.Color).
+                ClothingColor2(caster.Team.Color2).
+                TroopOrigin(troopOrigin).
+                CivilianEquipment(Mission.Current.DoesMissionRequireCivilianEquipment).
+                IsReinforcement(true).
+                SpawnOnInitialPoint(true).
+                InitialFrame(frame);
+            Agent troop = Mission.Current.SpawnAgent(buildData, false, 1);
+            troop.SetWatchState(Agent.WatchState.Alarmed);
+        }
```

Comment:
this method doesn't work properly, cause you can lose a battle if you die and have only the summoned skeletons, even if you have a lot of them